src/components/Wallet/NovaWallet/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { web3Accounts, web3Enable, web3FromAddress } from '@polkadot/extension-dapp';
+import { useCallback, useEffect, useState } from 'preact/compat';
+import { Modal } from 'react-daisyui';
+import { GlobalState } from '../../../GlobalStateProvider';
+import logo from '../../../assets/nova-wallet.png';
+import { PublicKey } from '../../PublicKey';
+
+export type NovaWalletProps = {
+  setWalletAccount: GlobalState['setWalletAccount'];
+};
+
+interface ExtensionAccount {
+  address: string;
+  name: string;
+  source: string;
+}
+
+const NovaWallet = ({ setWalletAccount }: NovaWalletProps) => {
+  const [openModal, setOpenModal] = useState<boolean>(false);
+  const [accounts, setAccounts] = useState<ExtensionAccount[]>([]);
+  const [selectedAccount, setSelectedAccount] = useState<ExtensionAccount>();
+
+  const onClick = useCallback(async () => {
+    const allInjected = await web3Enable('Pendulum Chain Portal');
+    const allAccounts = await web3Accounts();
+    setAccounts(
+      allAccounts
+        .filter(({ meta: { source } }) => source === 'polkadot-js')
+        .map(
+          ({ address, meta }): ExtensionAccount => ({
+            address: address,
+            name: meta.name || 'My Account',
+            source: meta.source,
+          }),
+        ),
+    );
+    setOpenModal(true);
+  }, [setOpenModal]);
+
+  useEffect(() => {
+    async function buildWalletAccount(extAcc: ExtensionAccount) {
+      const signer = await web3FromAddress(extAcc.address);
+      return {
+        address: extAcc.address,
+        source: extAcc.source,
+        name: extAcc.name,
+        signer,
+        wallet: {
+          enable: () => undefined,
+          extensionName: 'polkadot-js',
+          title: 'Nova Wallet',
+          installUrl: 'https://novawallet.io/',
+          logo: {
+            src: logo,
+            alt: 'Nova Wallet',
+          },
+          installed: true,
+          extension: undefined,
+          signer,
+          getAccounts: () => Promise.resolve([]), // TODO
+          subscribeAccounts: () => undefined, // TODO
+          transformError: (err: any) => err, // TODO
```

Comment:
Checked from the Nova dApp browser can confirm it works, Good to merge from my end.

Reply:
Good point @ebma , I think basically these functions are used internally by the Talisman modal and, since we are just creating the wallet object without using it inside their modal, then we don't need them. But correct me if I'm wrong.